<!DOCTYPE html
  PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>twisted.trial.reporter.TestResult.addUnexpectedSuccess : API documentation</title>
    <meta content="text/html;charset=utf-8" http-equiv="Content-Type" />
    <link href="apidocs.css" type="text/css" rel="stylesheet" />
    
    
  </head>
  <body>
    <h1 class="method">t.t.r.T.addUnexpectedSuccess(self, test, todo) : method documentation</h1>
    <p>
      <span id="part">Part of <a href="twisted.html">twisted</a>.<a href="twisted.trial.html">trial</a>.<a href="twisted.trial.reporter.html">reporter</a>.<a href="twisted.trial.reporter.TestResult.html">TestResult</a></span>
      <a href="http://twistedmatrix.com/trac/browser/tags/releases/twisted-11.0.0/twisted/trial/reporter.py#L137">View Source</a>
      
    </p>
    <div>
      
    </div>
    <div>Report that the given test succeeded against expectations.</p>
<p>In Trial, tests can be marked 'todo'. That is, they are expected to 
fail. When a test that is expected to fail instead succeeds, it should call
this method to report the unexpected success.<table class="fieldTable"><tr class="fieldStart"><td class="fieldName">Parameters</td><td class="fieldArg">test</td><td> (type: <code>pyunit.TestCase</code>
)</td></tr><tr><td></td><td class="fieldArg">todo</td><td> (type: <a href="twisted.trial.unittest.Todo.html"><code>unittest.Todo</code></a>
)</td></tr></table></div>

    
    
    <div id="splitTables">
      
      
      
    </div>
    
    
    

    
    <address>
      <a href="index.html">API Documentation</a> for <a href="http://twistedmatrix.com/">Twisted</a>, generated by <a href="http://codespeak.net/~mwh/pydoctor/">pydoctor</a> at 2011-04-03 00:32:03.
    </address>
  </body>
</html>